pytensor/tensor/blas.py

```diff
@@ -113,23 +113,22 @@
 _logger = logging.getLogger("pytensor.tensor.blas")
 
 
-# If check_init_y() == True we need to initialize y when beta == 0.
-def check_init_y():
-    # TODO: What is going on here?
+def gemv_impl_sensitive_to_unitialized_y():
+    # Check whether BLAS GEMV could output nan for unitialized y
     from scipy.linalg.blas import get_blas_funcs
 
-    if check_init_y._result is None:
-        y = float("NaN") * np.ones((2,))
+    if gemv_impl_sensitive_to_unitialized_y._result is None:
```

Comment:
I've never seen this `func._result` syntax before -- what is going on here, and why aren't we just doing something simpler?

Reply:
It's just caching the result so we don't do it every single time

Reply:
This is as simple as it gets, other than storing it in a global variable which is less elegant. LRU cache is overkill for this

Reply:
Understood; just never seen it before